clients/src/main/java/org/apache/kafka/common/record/Record.java

```diff
@@ -100,149 +103,17 @@ public Record(ByteBuffer buffer) {
     }
 
     // Package private constructor for inner iteration.
```

Comment:
Actually I had to remove these assertions. First, if the magic is 1, then we can't distinguish between a shallow uncompressed entry, and a deep entry, which would also not have its compression flag enabled. We could at least raise an error if magic is 0 and there is a provided wrapper timestamp., but we currently have test cases which allow us to create a Record with only a valid CRC (which means we can't check magic). We could probably be a little stricter: if there aren't enough bytes in the ByteBuffer to read the record overhead, then obviously the record can't be valid, so maybe those test cases are kind of silly and should be removed.